include/grpc/support/thd_id.h

```diff
@@ -16,15 +16,27 @@
  *
  */
 
-#ifndef GRPC_CORE_LIB_GPR_THD_INTERNAL_H
-#define GRPC_CORE_LIB_GPR_THD_INTERNAL_H
+#ifndef GRPC_SUPPORT_THD_ID_H
+#define GRPC_SUPPORT_THD_ID_H
+/** Thread interface for GPR.
 
-#include <grpc/support/time.h>
+   Types
+        gpr_thd_id        a unique opaque identifier for a thread.
```

Comment:
any special reason why we want to put this in a separate file?

Reply:
I would have liked to fully internalize the whole thing, but wrapped languages need thread-id for logging purposes, and they have to get it from `include/grpc`, not from `src/lib`. So this stuff stays behind in public headers but all the other contents can get internalized. C# in particular is using this function and type.

Reply:
ah Ok. got it. let's add this as a comment, so it's clear to future readers.

Reply:
Done. Thanks!